ee/observe/ingest_test.go

```
@@ -0,0 +1,207 @@
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/stretchr/testify/require"
+)
+
+// serveIngest routes a request through the handler ONLY (no middleware), so
+// handler-contract tests are not perturbed by the rate limiter or the app
+// resolver. The middleware chain has its own tests.
+func serveIngest(handler *IngestHandler, method, path string, body []byte) *httptest.ResponseRecorder {
+	router := mux.NewRouter()
+	router.HandleFunc("/observe/{APP_ID}/{PROJECT_ID}/v1/logs", handler.HandleLogs).Methods(http.MethodPost)
+	router.HandleFunc("/observe/{APP_ID}/{PROJECT_ID}/v1/metrics", handler.HandleMetrics).Methods(http.MethodPost)
+	req := httptest.NewRequest(method, path, bytes.NewReader(body))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)